### ISSUE TITLE
Loss function always return 0

### ISSUE THREAD

**Issue by hxue3** [nwojke/cosine_metric_learning#101]
I am running into an issue that the loss function always returns 0 while using triplet loss function. 

And if I use cosine-softmax, it won't return 0 but it will give me another error 
LossTensor is inf or nan : Tensor had NaN values

Any idea why
